modern/src/settings/MaintenancesPage.js

```diff
@@ -0,0 +1,91 @@
+import React, { useState } from 'react';
+import MainToolbar from '../MainToolbar';
+import { TableContainer, Table, TableRow, TableCell, TableHead, TableBody, makeStyles, IconButton } from '@material-ui/core';
+import MoreVertIcon from '@material-ui/icons/MoreVert';
+import t from '../common/localization';
+import { useEffectAsync } from '../reactHelper';
+import EditCollectionView from '../EditCollectionView';
+
+import positionAttributes from '../attributes/positionAttributes';
+import { formatDistance, formatSpeed } from '../common/formatter';
+import { useAttributePreference } from '../common/preferences';
+
+const useStyles = makeStyles(theme => ({
+  columnAction: {
+    width: theme.spacing(1),
+    padding: theme.spacing(0, 1),
+  },
+}));
+
+const MaintenancesView = ({ updateTimestamp, onMenuClick }) => {
+  const classes = useStyles();
+
+  const [items, setItems] = useState([]);
+  const speedUnit = useAttributePreference('speedUnit');
+  const distanceUnit = useAttributePreference('distanceUnit');
+
+  useEffectAsync(async () => {
+    const response = await fetch('/api/maintenance');
+    if (response.ok) {
+      setItems(await response.json());
+    }
+  }, [updateTimestamp]);
+
+  const convertAttribute = (key, value) => {
```

Comment:
Don't we have something similar already?

Reply:
I didn't find anything.